window_management/window_tweak.py

```diff
@@ -0,0 +1,525 @@
+testing: bool = True
+
+win_compass_control = None
+compass_control = None
+ctx_stop = None
+
+# talon stuff
+
+mod = Module()
+
+TAG_NAME = 'window_tweak_running'
+tag = mod.tag(TAG_NAME, desc="Enable stop command during continuous window move/resize.")
+
+# context used to enable/disable window_tweak_running tag
+ctx = Context()
+
+setting_move_frequency = mod.setting(
+    "win_move_frequency",
+    type=str,
+    default="40ms",
+    desc="The update frequency used when moving a window continuously",
+)
+setting_resize_frequency = mod.setting(
+    "win_resize_frequency",
+    type=str,
+    default="40ms",
+    desc="The update frequency used when resizing a window continuously",
+)
+setting_move_rate = mod.setting(
+    "win_continuous_move_rate",
+    type=float,
+    default=4.5,
+    desc="The target speed, in cm/sec, for continuous move operations",
+)
+setting_resize_rate = mod.setting(
+    "win_continuous_resize_rate",
+    type=float,
+    default=4.0,
+    desc="The target speed, in cm/sec, for continuous resize operations",
+)
+mod.setting(
+    "win_hide_move_gui",
+    type=int,
+    default=0,
+    desc="When enabled, the 'Move/Resize Window' GUI will not be shown for continuous move operations.",
+)
+mod.setting(
+    "win_hide_resize_gui",
+    type=int,
+    default=0,
+    desc="When enabled, the 'Move/Resize Window' GUI will not be shown for continuous resize operations.",
+)
+mod.setting(
+    "win_set_queue_timeout",
+    type=float,
+    default=0.2,
+    desc="How long to wait (in seconds) for talon to signal completion of window move/resize requests.",
+)
+mod.setting(
+    "win_set_retries",
+    type=int,
+    default=1,
+    desc="How many times to retry a timed out talon window move/resize request.",
+)
+setting_verbose_warnings = mod.setting(
+    "win_verbose_warnings",
+    type=bool,
+    default=False,
+    # window move and resize requests are not guaranteed
+    desc="Whether to generate warnings for anomalous events.",
+)
+
+@imgui.open(y=0)
+def win_stop_gui(gui: imgui.GUI) -> None:
+    gui.text(f"Say 'window stop' or click below.")
+    gui.line()
+    if gui.button("Stop moving/resizing"):
+        actions.user.win_stop()
+
+@imgui.open(x=2100, y=40)
+# @imgui.open(x=4000,y=244)
+def _win_show_gui(gui: imgui.GUI) -> None:
+    w = ui.active_window()
+
+    gui.text(f"== Window ==")
+
+    gui.text(f"Id: {w.id}")
+    gui.spacer()
+
+    x = w.rect.x
+    y = w.rect.y
+    width = w.rect.width
+    height = w.rect.height
+
+    gui.text(f"Top Left: {x, y}")
+    gui.text(f"Top Right: {x + width, y}")
+    gui.text(f"Bottom Left: {x, y + height}")
+    gui.text(f"Bottom Right: {x + width, y + height}")
+    gui.text(f"Center: {round(w.rect.center.x), round(w.rect.center.y)}")
+    gui.spacer()
+
+    gui.text(f"Width: {round(width)}")
+    gui.text(f"Height: {round(height)}")
+
+    gui.line()
+
+    gui.text(f"== Mouse ==")
+
+    gui.text(f"Position: {ctrl.mouse_pos()}")
```

Comment:
On Mac, ctrl.mouse_pos() will provide floats. We'll probably want to format this to 2 decimals places or something. Otherwise, the window will probably be unusable on Mac as the mouse moves